larq_compute_engine/tflite/kernels/bconv2d.cc

```diff
@@ -55,6 +55,10 @@ typedef struct {
   // padding
   TfLitePadding padding_type{};
   TfLitePaddingValues padding_values{};
+  // We could copy `TfLitePadding` to include an extra value kTfLitePaddingOne.
+  // However this requires copying a lot of other code like im2col, so it is
+  // easier to simply have one extra flag here.
+  bool one_padding = false;
```

Comment:
I tried that first (I have the code in a separate branch) but this requires copying a lot of tflite code such as the im2col algorithm. It became quite involved, and in my opinion it was less readable than this. However I'm still open to this idea, let me know what you think.

Reply:
why do you need to copy im2col if you subclass the struct?

Reply:
So there is `TfLitePadding` (defined in [builtin_op_data.h](https://github.com/tensorflow/tensorflow/blob/844b64f2eed4fb6ddc7e893d8baba0e35abd16dd/tensorflow/lite/c/builtin_op_data.h#L45) ) which is an `enum`.
This is used by `ComputePaddingHeightWidth` in `Prepare`.

Then inside `ConvParams` there is `PaddingType padding_type;` . This is a `class` defined in [types.h](https://github.com/tensorflow/tensorflow/blob/844b64f2eed4fb6ddc7e893d8baba0e35abd16dd/tensorflow/lite/kernels/internal/types.h#L28).

During `EvalOpt` we convert the `TfLitePadding` value to the `PaddingType` value in `ConvParams`.

So I tested the following options:

- Copy `TfLitePadding` to `LcePadding` and `PaddingType` to `LcePaddingType` . For both cases, add the extra option for one-padding.
*Pros*: we have a proper third padding type everywhere
*Cons*: because we change `PaddingType` to `LcePaddingType` we also have to copy `ConvParams` to `BConvParams`. Since im2col accepts only `ConvParams` we have to copy im2col to accept `BConvParams`.

- Change our `TfLiteBConv2DParams` to subclass the original `ConvParams`
  The original `ConvParams` has the `PaddingType` which we can not set to one-padding. Therefore, in our subclass we add `LcePadding true_padding_type;` which has support for one-padding.
*Pros*: We no longer have to copy to `ConvParams` every `Eval` step, because our class is already a `ConvParams` subclass. I benchmarked it but it does not provide a measurable speedup, but I do think its cleaner because it avoids the copy.
This does not require copying im2col or whatever so thats pretty nice.
*Cons*: its a bit confusing because then we have `params->padding_type` (from the base class) which can only be valid or same, and we have `params->true_padding_type` which can also be One.
Furthermore, when you call `ComputePaddingHeightWidth` you have to still convert `true_padding_type` back to `TfLitePaddingType` and send `ONE` to `SAME` to get the right output size.


Actually I liked this second option a lot, but it was a pretty huge diff compared to master (because things like `dilations` now moved from our `conv_params` into the base class). The current PR is a much smaller and more readable diff.

How about this: I first push a PR where i make our struct a subclass of ConvParams and then I push another PR where I add the one-padding on top of that.

Reply:
> So there is `TfLitePadding` (defined in [builtin_op_data.h](https://github.com/tensorflow/tensorflow/blob/844b64f2eed4fb6ddc7e893d8baba0e35abd16dd/tensorflow/lite/c/builtin_op_data.h#L45) ) which is an `enum`.
> This is used by `ComputePaddingHeightWidth` in `Prepare`.
> 
> Then inside `ConvParams` there is `PaddingType padding_type;` . This is a `class` defined in [types.h](https://github.com/tensorflow/tensorflow/blob/844b64f2eed4fb6ddc7e893d8baba0e35abd16dd/tensorflow/lite/kernels/internal/types.h#L28).
> 
> During `EvalOpt` we convert the `TfLitePadding` value to the `PaddingType` value in `ConvParams`.

Both `PaddingType` and `TfLitePadding` are `enums` so you can not inhert them but you can use the trick explained here: https://www.codeproject.com/Articles/16150/Inheriting-a-C-enum-type